PersonApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -46,14 +52,22 @@ public async Task<Person> PostNewPersonAsync(CreatePersonRequestObject requestOb
         [LogCall]
         public async Task<UpdateEntityResult<PersonDbEntity>> UpdatePersonByIdAsync(UpdatePersonRequestObject requestObject, string requestBody, PersonQueryObject query)
         {
-            _logger.LogDebug($"Calling IDynamoDBContext.SaveAsync to update id {query.Id}");
-
             var existingPerson = await _dynamoDbContext.LoadAsync<PersonDbEntity>(query.Id).ConfigureAwait(false);
             if (existingPerson == null) return null;
 
+            // Check the If-Match header value
+            string ifMatch = _httpContextAccessor.HttpContext.Request.Headers.GetHeaderValue(HeaderConstants.IfMatch);
```

Comment:
I am hesitant to introduce a dependency on the http request in the gateway. I would think you'd pass the version number in, and get it from the header in the controller...

Reply:
Fair point.